movement/watch_faces/complication/databank_face.c

```diff
@@ -40,12 +41,99 @@ const char *pi_data[] = {
 
 const int databank_num_pages = (sizeof(pi_data) / sizeof(char*) / 2);
 
+typedef enum {
+    DATABANK_DISPLAY = 0,
+    DATABANK_CHIRPING,
+} databank_mode_t;
+
 struct {
     uint8_t current_word;
     uint8_t databank_page;
     bool animating;
+
+    // Current operating mode, used to know if we are chirping or not.
+    databank_mode_t mode;
+
+    // Helps us handle 1/64 ticks during chirpy transmission; including countdown timer
+    chirpy_tick_state_t tick_state;
+
+    // Used by chirpy encoder during transmission
+    chirpy_encoder_state_t encoder_state;
 } databank_state;
 
+// Support for chirping out the data. See chipy_demo_face.* for more information
+static void _databank_quit_chirping() {
+    databank_state.mode = DATABANK_DISPLAY;
+    watch_set_buzzer_off();
+    watch_clear_indicator(WATCH_INDICATOR_BELL);
+    movement_request_tick_frequency(1);
+}
+
+static void _databank_data_tick(void *context) {
+    uint8_t tone = chirpy_get_next_tone(&databank_state.encoder_state);
+    // Transmission over?
+    if (tone == 255) {
+        _databank_quit_chirping();
+        return;
+    }
+    uint16_t period = chirpy_get_tone_period(tone);
+    watch_set_buzzer_period(period);
+    watch_set_buzzer_on();
+}
+
+// Which byte are we currently processing?
+static uint16_t curr_data_ix;
+
+static uint8_t _databank_get_next_byte(uint8_t *next_byte) {
+    // We always send out the currently displayed dataset
+    int curr_pos = databank_state.databank_page * 2 + 1;
```

Comment:
I'll be honest I don't quite understand the pagination used into the pi_data array, but this seems to work.